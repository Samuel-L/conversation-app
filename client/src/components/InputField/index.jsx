import React from 'react';
import PropTypes from 'prop-types';

import { withStyles } from '@material-ui/core/styles';
import FormControl from '@material-ui/core/FormControl';
import InputLabel from '@material-ui/core/InputLabel';
import Input from '@material-ui/core/Input';
import FormHelperText from '@material-ui/core/FormHelperText';

const styles = theme => ({
  formControl: {
    margin: theme.spacing.unit,
  },
});

export const InputField = (props) => {
  const { classes } = props;

  return (
    <FormControl error={props.error} className={classes.formControl}>
      <InputLabel htmlFor={props.id}>{props.label}</InputLabel>
      <Input
        id={props.id}
        type={props.type}
        value={props.value}
        onChange={props.handleChange}
      />
      <FormHelperText id={`${props.id}-help-text`}>
        { props.error ? props.errorMessage : null }
      </FormHelperText>
    </FormControl>
  );
};


InputField.propTypes = {
  error: PropTypes.bool,
  errorMessage: PropTypes.string,
  id: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  handleChange: PropTypes.func.isRequired,
  classes: PropTypes.objectOf(PropTypes.string).isRequired,
};

InputField.defaultProps = {
  error: false,
  errorMessage: '',
};

export default withStyles(styles)(InputField);
