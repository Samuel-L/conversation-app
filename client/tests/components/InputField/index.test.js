import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import InputField from '../../../src/components/InputField';

describe('component: InputField', () => {
  const classes = { formControl: '' };

  describe('rendering', () => {
    const tree = renderer.create(<InputField
      id="test"
      label="test"
      type="text"
      classes={classes}
      value=""
      handleChange={jest.fn()}
    />).toJSON();

    it('matches snapshot', () => {
      expect(tree).toMatchSnapshot();
    });
  });
});
