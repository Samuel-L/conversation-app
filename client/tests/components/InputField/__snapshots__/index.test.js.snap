// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component: InputField rendering matches snapshot 1`] = `
<div
  className="MuiFormControl-root-2 InputField-formControl-1"
  onBlur={[Function]}
  onFocus={[Function]}
>
  <label
    className="MuiFormLabel-root-11 MuiInputLabel-root-6 MuiInputLabel-formControl-7 MuiInputLabel-animated-10"
    data-shrink={false}
    htmlFor="test"
  >
    test
  </label>
  <div
    className="MuiInput-root-16 MuiInput-formControl-17 MuiInput-underline-20"
  >
    <input
      aria-invalid={false}
      aria-required={false}
      className="MuiInput-input-24"
      disabled={false}
      id="test"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      type="text"
      value=""
    />
  </div>
  <p
    className="MuiFormHelperText-root-29"
    id="test-help-text"
  />
</div>
`;
